CI failed due to unrelated bug:
```
2023-02-03T13:55:36.8225525Z XPASS tests/unit/api/usecases/exportable_code/test_streamer.py::TestStreamer::test_threaded_streamer CVS-102619
2023-02-03T13:55:36.8226585Z FAILED tests/integration/cli/detection/test_instance_segmentation.py::TestInstanceSegmentationCLI::test_nncf_optimize[Custom_Counting_Instance_Segmentation_MaskRCNN_EfficientNetB2B]
2023-02-03T13:55:36.8227496Z = 1 failed, 831 passed, 6 skipped, 2 xpassed, 35 warnings in 5266.59s (1:27:46) =
2023-02-03T13:55:38.3652729Z ERROR: InvocationError for command /home/validation/actions-runner/_work/training_extensions/training_extensions/.tox/pre-merge/bin/coverage run '--include=otx/*' -m pytest -ra --showlocals tests/unit tests/integration/cli (exited with code 1)
```
BTW locally I also observe this problem when launching all integration tests.

--- a/otx/algorithms/classification/adapters/mmcls/data/datasets.py
+++ b/otx/algorithms/classification/adapters/mmcls/data/datasets.py
@@ -191,21 +191,18 @@ class MPAMultilabelClsDataset(MPAClsDataset):
         """Load annotations."""
         include_empty = self.empty_label in self.labels
         for i, _ in enumerate(self.otx_dataset):
-            class_indices = []
             item_labels = self.otx_dataset[i].get_roi_labels(self.labels, include_empty=include_empty)
             ignored_labels = self.otx_dataset[i].ignored_labels
+            onehot_indices = np.zeros(len(self.labels))
             if item_labels:
                 for otx_lbl in item_labels:
                     if otx_lbl not in ignored_labels:
-                        class_indices.append(self.label_names.index(otx_lbl.name))
+                        onehot_indices[self.label_names.index(otx_lbl.name)] = 1
                     else:
-                        class_indices.append(-1)
-            else:  # this supposed to happen only on inference stage or if we have a negative in multilabel data
-                class_indices.append(-1)
-            onehot_indices = np.zeros(len(self.labels))
-            for idx in class_indices:
-                if idx != -1:  # TODO: handling ignored label?
-                    onehot_indices[idx] = 1
+                        # during trainign we filter ignored classes out,
+                        # during validation mmcv's mAP also filters -1 labels
+                        onehot_indices[self.label_names.index(otx_lbl.name)] = -1
+
             self.gt_labels.append(onehot_indices)
         self.gt_labels = np.array(self.gt_labels)
 

--- a/otx/algorithms/classification/tasks/inference.py
+++ b/otx/algorithms/classification/tasks/inference.py
@@ -105,7 +105,7 @@ class ClassificationInferenceTask(
             logger.info("Classification mode: hierarchical")
             self._hierarchical = True
             self._hierarchical_info = get_hierarchical_info(task_environment.label_schema)
-        else:
+        if not self._multilabel and not self._hierarchical:
             logger.info("Classification mode: multiclass")
 
         if self._hyperparams.algo_backend.train_type == TrainType.SELFSUPERVISED:

--- a/otx/mpa/modules/models/heads/custom_multi_label_linear_cls_head.py
+++ b/otx/mpa/modules/models/heads/custom_multi_label_linear_cls_head.py
@@ -65,8 +65,13 @@ class CustomMultiLabelLinearClsHead(MultiLabelClsHead):
         img_metas = kwargs.get("img_metas", False)
         gt_label = gt_label.type_as(x)
         cls_score = self.fc(x) * self.scale
+
+        valid_batch_mask = gt_label >= 0
+        gt_label = gt_label[valid_batch_mask,].view(gt_label.shape[0], -1)
+        cls_score = cls_score[valid_batch_mask,].view(cls_score.shape[0], -1)
         if img_metas:
             valid_label_mask = self.get_valid_label_mask(img_metas=img_metas)
+            valid_label_mask = valid_label_mask[valid_batch_mask,].view(valid_label_mask.shape[0], -1)
             valid_label_mask = valid_label_mask.to(x.device)
             losses = self.loss(cls_score, gt_label, valid_label_mask=valid_label_mask)
         else:

--- a/otx/mpa/modules/models/heads/custom_multi_label_non_linear_cls_head.py
+++ b/otx/mpa/modules/models/heads/custom_multi_label_non_linear_cls_head.py
@@ -93,8 +93,13 @@ class CustomMultiLabelNonLinearClsHead(MultiLabelClsHead):
         img_metas = kwargs.get("img_metas", False)
         gt_label = gt_label.type_as(x)
         cls_score = self.classifier(x) * self.scale
+
+        valid_batch_mask = gt_label >= 0
+        gt_label = gt_label[valid_batch_mask,].view(gt_label.shape[0], -1)
+        cls_score = cls_score[valid_batch_mask,].view(cls_score.shape[0], -1)
         if img_metas:
             valid_label_mask = self.get_valid_label_mask(img_metas=img_metas)
+            valid_label_mask = valid_label_mask[valid_batch_mask,].view(valid_label_mask.shape[0], -1)
             valid_label_mask = valid_label_mask.to(x.device)
             losses = self.loss(cls_score, gt_label, valid_label_mask=valid_label_mask)
         else:

